app/models/miq_request_workflow.rb

```diff
@@ -1042,7 +1042,8 @@ def ems_folder_to_hash_struct(ci)
   end
 
   def storage_to_hash_struct(ci)
-    build_ci_hash_struct(ci, [:name, :free_space, :total_space, :storage_domain_type])
+    storage_cluster = ci.storage_clusters.detect { |cluster| cluster.ems_id == get_source_and_targets[:ems].try(:id) }
+    build_ci_hash_struct(ci, [:name, :free_space, :total_space, :storage_domain_type]).tap { |hs| hs.storage_cluster = storage_cluster.try(:name) }
```

Comment:
Actually it's failing the test after I moved get_source_and_targets out of the block,
```
def storage_to_hash_struct(ci)
    ems_id = get_source_and_targets[:ems].try(:id)
    storage_cluster = ci.storage_clusters.detect { |cluster| cluster.ems_id == ems_id }
    build_ci_hash_struct(ci, [:name, :free_space, :total_space, :storage_domain_type]).tap { |hs| hs.storage_cluster = storage_cluster.try(:name) }
  end
```
Test fails with 
```
     Failure/Error: ems_id = get_source_and_targets[:ems].try(:id)

     NameError:
       undefined local variable or method `get_source_and_targets' for #<MiqProvisionWorkflow:0x007f85192db8d8>
```

Any clue?